### PR TITLE
menu overlaps when clicking on collapse

### LINKED ISSUE #37: Settings Menu overlaps
If you open the settings menu and minimize the navigationbar the inner from settings is not change the position

image

image

### PR DESCRIPTION
**Pull request by stgeipel**
Replaced the menuhandleclick function with a collapse state.
The menu no longer overlaps and is displayed with a hover

closes #37

### CODE REVIEW

**Comment by madisvain** [comment]
@stgeipel thank you very much for this pull request 👍. I will review it latest tomorrow.

File: src/components/layout/navigation.js [hunk 1]
```diff
@@ -22,48 +22,15 @@ class Navigation extends Component {
...
           </Link>
         </div>
-        <Menu
-          theme="dark"
-          mode="inline"
-          selectedKeys={selectedMenuKeys}
-          openKeys={openKeys}
-          onClick={this.handleMenuClick}
-        >
+        <Menu theme="dark" mode="inline" defaultSelectedKeys={['1']}>
```
Comment (madisvain) [inline src/components/layout/navigation.js]: There is no key with string '1' so the defaultSelectedKeys could be 'invoices'.

**Comment by madisvain** [comment]
@stgeipel the main question is that currently the solution fixes the bug. Thats true and the code is super simple but it looses the state after a refresh. And it does not know to keep the submenu open after a refresh.

That in essence is not an issue while running an Electron app because nobody is supposed to refresh (but it's possible to do). But I have an idea about deploying the same codebase online where this would be a fequent thing.

So maybe we can figure out how to keep your simplification but have it get the default key correct when reloading and also keeping the submenu open when it needs to.

**Comment by madisvain** [comment]
@stgeipel also by the way. I'm very very grateful for all the feedback and this pull request. You have motivated me a lot to keep this project going. I'd like to add a CONTRIBUTORS file to this repo also.

Thank you! 👍

**Comment by stgeipel** [comment]
@madisvain Thanks for the feedback. I am already working on a solution that the state is preserved and the submenu remains open.
I am glad to hear that you want to keep going the project. 👍 
I would be happy about the CONTRIBUTORS file and will try to help as much as possible.

File: src/components/layout/navigation.js (hunk 1 above)
Comment (stgeipel) [reply src/components/layout/navigation.js]: ja du hast. Behebe ich mit dem nächsten Request

**Comment by stgeipel** [comment]
I m fixing the problem by using the localStorage. If you open an submenu and refreshing the app it opens the last selected menutab.

If you close the Browsertab or the window you will be asked to select the organization. At this moment i clear the localStorage item because i dont want to open the last session.

File: src/components/layout/navigation.js (hunk 1 above)
Comment (stgeipel) [reply src/components/layout/navigation.js]: fixed by selecting from localStorage

**Comment by madisvain** [comment]
@stgeipel I reviewed the codebase and I like the cleanup a lot from the switch. But I found issues with the localStore usage - if you get a link from a friend then it's not initializing correctly because the localStore is empty for you until you click on a link.

I'm a big fan of storing the state of the route in the URL. That way the links always stay sharable and work out of the box. So I made another refactoring based on what you made. Maybe you can take a look and can merge this logic into your branch and we could push it live today?

https://github.com/madisvain/upcount/pull/40/commits/f6c854335a6512e9e929602a591d173ae22fdc97

Let me know what you think. I'll also be adding the CONTRIBUTORS file shortly and looking for a release tonight.

The multilingual branch is also ready to go. I just need to test it a bit.
https://github.com/madisvain/upcount/pull/39

**Comment by stgeipel** [comment]
@madisvain i think the solution is great. I have adjusted my code to your URL solution. 
So we don't need to save the state in the localStorage 👍

**Pull request merged**